### PR TITLE
feat(toggle group): add tokens

### LINKED ISSUE #6029: Tokenize toggle group

### PR DESCRIPTION
**Pull request by srambach**
Fixes #6029

### CODE REVIEW

**Review by andrew-ronaldson — APPROVED** [review]
Matches Figma, LGTM for design!

**Review by mattnolting — CHANGES_REQUESTED** [review]
Outstanding! 👏👏👏 

Looks like you can drop 

[& + & {
  margin-inline-start: var(--#{$toggle-group}__item--item--MarginLeft);
}
](https://github.com/patternfly/patternfly/pull/6190/files#diff-045989b534efcabcb44e8d2fea279cb9b1242530cc45366f9980aee247064568R60-R63) per the design spec 🎉 

Screenshot 2024-01-06 at 9 57 33 AM

**Comment by mattnolting** [comment]
@andrew-ronaldson As a followup, we need to consider mobile presentation. Here's an example of toggle group items overflowing and how `toggle-group` can handle this scenario. 

Screenshot 2024-01-06 at 11 04 36 AM

**Things to consider**
- [ ] Adjacent selected item border. When textual content is less consistent, adjacent selected items appear run together. Should we include border updates here??
- [ ] When toggle items extend past toggle group boundaries, the overflow is currently hidden (default). If we move the border to `toggle-group` instead of items, we can drop top/bottom borders on individual items, and conditionally drop left/right border for `:first/:last-child`. This results in the content being scrollable while presenting design accurately. 
- [ ] When `toggle-group` is a child of `display: flex/grid` its width grows. Should add `width: min-content` to `toggle-group` as well as `white-space: nowrap` to `toggle-group__button`. 

**Toggle group as a child of `display: flex/grid`**

Screenshot 2024-01-06 at 11 35 32 AM

**Toggle group button set to `white-space: nowrap`**

Screenshot 2024-01-06 at 11 33 30 AM



https://codesandbox.io/p/sandbox/infallible-meadow-msz34w?file=%2Fstyle.css%3A12%2C111

File: src/patternfly/components/ToggleGroup/toggle-group.scss [hunk 1]
```diff
@@ -1,50 +1,50 @@
...
+  --#{$toggle-group}__button--before--BorderWidth: var(--pf-t--global--border--width--control--default);
+  --#{$toggle-group}__button--before--BorderColor: var(--pf-t--global--border--color--default);
 
   // item
-  --#{$toggle-group}__item--item--MarginLeft: calc(-1 * var(--#{$pf-global}--BorderWidth--sm));
-  --#{$toggle-group}__item--first-child__button--BorderTopLeftRadius: var(--#{$pf-global}--BorderRadius--sm);
-  --#{$toggle-group}__item--first-child__button--BorderBottomLeftRadius: var(--#{$pf-global}--BorderRadius--sm);
-  --#{$toggle-group}__item--last-child__button--BorderTopRightRadius: var(--#{$pf-global}--BorderRadius--sm);
-  --#{$toggle-group}__item--last-child__button--BorderBottomRightRadius: var(--#{$pf-global}--BorderRadius--sm);
+  --#{$toggle-group}__item--item--MarginLeft: calc(-1 * var(--pf-t--global--border--width--control--default));
```
[the remark this answers is not in the document]
Comment (srambach) [reply src/patternfly/components/ToggleGroup/toggle-group.scss:24]: @mattnolting Actually I checked with @andrew-ronaldson and they'd still like to collapse that down to 1px; it was just too complicated in Figma.

**Review by mattnolting — APPROVED** [review]
LPTM!

**Comment by srambach** [comment]
Issue created to look at the mobile view as @mattnolting suggested. https://github.com/patternfly/patternfly/issues/6202

**Comment by lboehling** [comment]
Looks good! I also +1 nolting's comment about including a border so that adjacent clicked items don't look like they're running together.

**Review by kmcfaul — APPROVED** [review]

**Review by thatblindgeye — APPROVED** [review]

**Pull request merged**